React/Base/RCTComponentEvent.m

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "RCTComponentEvent.h"
+
+#import "RCTAssert.h"
+
+@implementation RCTComponentEvent
+{
+  NSArray *_arguments;
+}
+
+@synthesize eventName = _eventName;
+@synthesize viewTag = _viewTag;
+
+- (instancetype)initWithName:(NSString *)name body:(NSDictionary *)body
+{
+  if (self = [super init]) {
+    NSNumber *target = body[@"target"];
```

Comment:
Why should we extract target from body? Should we pass it directly?

Reply:
I think it is because target needs to be in the event body for JS so instead of putting it both in the body and passing it as an arg we just extract it from the body.

Reply:
Mmm... So, it means that our API reflects details of internal implementation? Does not sound cool. No?
Do we modify body somewhere down the road before it does to serialization process? If yes, that coupling `target` and `body` cannot be counted as optimization and we have to fix our API.

Reply:
We could probably pass target as a parameter and add it to the body in the init.

Reply:
Yes, I think so. We have to know for sure should we do this or not. If it is natural performancewise, I would do it.

Reply:
I thought the extra copy to make the NSDictionary mutable would be bad but we actually did it already here https://github.com/facebook/react-native/pull/15894/files#diff-b9fc78351b7b976a36356ef6f08c1f79L117. So the main difference performance wise is the extra allocation of RCTEvent